File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;

public static class SecretHandshake
{
    private static readonly List<Action<List<string>>> Operations = new()
    {
        commands => commands.Add("wink"),
        commands => commands.Add("double blink"),
        commands => commands.Add("close your eyes"),
        commands => commands.Add("jump"),
        commands => commands.Reverse()
    };

    public static IEnumerable<string> Commands(int code)
    {
        var commands = new List<string>();

        for (var i = 0; i < Operations.Count; i++)
            if ((code & 1 << i) != 0)
                Operations[i](commands);

        return commands;
    }
}

File: clojure/secret-handshake/src/secret_handshake.clj
(ns secret-handshake)

(def coded-commands
  {
     0 #(conj % "wink")
     1 #(conj % "double blink")
     2 #(conj % "close your eyes")
     3 #(conj % "jump")
     4 reverse
   })

(defn- update-commands [code commands [command-nth-bit command]]
  (if (bit-test code command-nth-bit) (command commands) commands))

(defn commands [code]
  (reduce (partial update-commands code) [] coded-commands))
